lib/runtime-core/src/instance.rs

```diff
@@ -814,6 +844,134 @@ impl<'a> DynFunc<'a> {
     }
 }
 
+impl<'a> Exportable<'a> for Memory {
+    fn get_self(exports: &'a Exports, name: &str) -> ResolveResult<Self> {
+        let (inst_inner, module) = exports.get_inner();
+        let export_index =
+            module
+                .info
+                .exports
+                .get(name)
+                .ok_or_else(|| ResolveError::ExportNotFound {
+                    name: name.to_string(),
+                })?;
+        if let ExportIndex::Memory(idx) = export_index {
+            Ok(inst_inner.get_memory_from_index(module, *idx))
+        } else {
+            Err(ResolveError::ExportWrongType {
+                name: name.to_string(),
+            })
+        }
+    }
+}
+
+impl<'a> Exportable<'a> for Table {
+    fn get_self(exports: &'a Exports, name: &str) -> ResolveResult<Self> {
+        let (inst_inner, module) = exports.get_inner();
+        let export_index =
+            module
+                .info
+                .exports
+                .get(name)
+                .ok_or_else(|| ResolveError::ExportNotFound {
+                    name: name.to_string(),
+                })?;
+        if let ExportIndex::Table(idx) = export_index {
+            Ok(inst_inner.get_table_from_index(module, *idx))
+        } else {
+            Err(ResolveError::ExportWrongType {
+                name: name.to_string(),
+            })
+        }
+    }
+}
+
+impl<'a> Exportable<'a> for Global {
+    fn get_self(exports: &'a Exports, name: &str) -> ResolveResult<Self> {
+        let (inst_inner, module) = exports.get_inner();
+        let export_index =
+            module
+                .info
+                .exports
+                .get(name)
+                .ok_or_else(|| ResolveError::ExportNotFound {
+                    name: name.to_string(),
+                })?;
+        if let ExportIndex::Global(idx) = export_index {
+            Ok(inst_inner.get_global_from_index(module, *idx))
+        } else {
+            Err(ResolveError::ExportWrongType {
+                name: name.to_string(),
+            })
+        }
+    }
+}
+
+impl<'a> Exportable<'a> for DynFunc<'a> {
+    fn get_self(exports: &'a Exports, name: &str) -> ResolveResult<Self> {
+        let (inst_inner, module) = exports.get_inner();
+        dyn_func(module, inst_inner, name)
```

Comment:
I think it will be cleared to move `dyn_func` contents into here? (unless dyn_func is used somewhere else)